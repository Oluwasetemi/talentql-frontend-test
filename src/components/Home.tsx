import * as React from 'react';
import { AppWrapper } from '../App';
import { AuthButton } from './AuthButton';
import { ColorsFilter } from './Colors';
import { Nav } from './Nav';
import { RenderBox } from './RenderBox';
import { ShapesFilter } from './Shapes';

function ColorDashboard() {
	return (
		<>
			<Nav />
			<div className="App-body">
				<AppWrapper>
					<AuthButton />
					<h3>Filters</h3>
					<h5 className="blue-text">Shapes</h5>
					<ShapesFilter />
					<h5 className="blue-text">Color.colors</h5>
					<ColorsFilter />
					<RenderBox />
				</AppWrapper>
			</div>
		</>
	);
}

export default ColorDashboard;
