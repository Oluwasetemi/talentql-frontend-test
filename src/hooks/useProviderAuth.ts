import React from 'react';
import { fakeAuth } from '../App';

export function useProvideAuth() {
	const [user, setUser] = React.useState<string | null>(null);

	const signin = (cb: () => void) => {
		return fakeAuth.signin(() => {
			setUser('user');
			cb();
		});
	};

	const signout = (cb: () => void) => {
		return fakeAuth.signout(() => {
			setUser(null);
			cb();
		});
	};

	return {
		user,
		signin,
		signout,
	};
}
