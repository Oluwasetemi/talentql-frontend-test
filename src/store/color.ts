// create the color slice with it reducers
// checkColor
// unCheckColor
// restoreColors
import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import type { IColors } from '../App';
import { colors } from '../filter.json';

export type colorState = {
	colors: Array<IColors>;
	selectedColors: Array<string>;
};

const colorTypes = colors.map((color) => color.color);

export const initialState: colorState = {
	colors,
	selectedColors: [...colorTypes],
};

// function isChecked(selectedColors: string[], color: string) {
// 	return selectedColors.includes(color);
// }

export const colorSlice = createSlice({
	name: 'Colors',
	initialState,
	reducers: {
		isChecked(colors, action) {
			colors.selectedColors.includes(action.payload);
		},
		checkColor(colors, action: PayloadAction<string>) {
			colors.selectedColors.push(action.payload);
		},
		unCheckColor(colors, action) {
			// console.log(action);
			const index = colors.selectedColors.findIndex(
				(color) => color === action.payload,
			);
			colors.selectedColors.splice(index, 1);
			// console.log(a);
		},
		resetColors(colors) {
			colors.selectedColors = colorTypes;
		},
	},
});

export const { isChecked, checkColor, unCheckColor, resetColors } =
	colorSlice.actions;
export default colorSlice.reducer;
