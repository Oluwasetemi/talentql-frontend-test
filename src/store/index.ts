import { configureStore } from '@reduxjs/toolkit';
import colorReducer from './color';
import shapeReducer from './shape';

const store = configureStore({
	reducer: {
		color: colorReducer,
		shape: shapeReducer,
	},
});
export default store;

export type RootState = ReturnType<typeof store.getState>;

export type AppDispatch = typeof store.dispatch;
