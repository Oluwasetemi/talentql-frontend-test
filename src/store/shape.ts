// create the color slice with it reducers
// checkColor
// unCheckColor
// restoreColors
import { createSlice } from '@reduxjs/toolkit';
import { shapes } from '../filter.json';

export type shapeState = {
	shapes: typeof shapes;
	selectedShapes: typeof shapes;
};
const initialState: shapeState = {
	shapes,
	selectedShapes: shapes,
};

export const shapeSlice = createSlice({
	name: 'Shapes',
	initialState,
	reducers: {
		checkShape: (shapes, action) => {},
		unCheckShape: (shapes, action) => {},
		restoreShapes: (shapes, action) => {},
	},
});

export const { checkShape, unCheckShape, restoreShapes } = shapeSlice.actions;
export default shapeSlice.reducer;
