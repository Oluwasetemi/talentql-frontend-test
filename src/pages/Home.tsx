import * as React from 'react';
import { AuthButton } from '../components/AuthButton';
import { ColorsFilter } from '../components/Colors';
import { Navigation } from '../components/Navigation';
import { RenderBox } from '../components/RenderBox';
import { ShapesFilter } from '../components/Shapes';

function Dashboard() {
	return (
		<Navigation>
			<AuthButton />
			<h3>Filters</h3>
			<h5 className="blue-text">Shapes</h5>
			<ShapesFilter />
			<h5 className="blue-text">Colors</h5>
			<ColorsFilter />
			<RenderBox />
		</Navigation>
	);
}

export default Dashboard;
