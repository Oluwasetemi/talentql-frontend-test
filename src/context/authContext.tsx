import React from 'react';
import { useProvideAuth } from '../hooks/useProviderAuth';

export interface AppContextInterface {
	user: string | null;
	signin: (cb: () => void) => void;
	signout: (cb: () => void) => void;
}

export const authContext = React.createContext<AppContextInterface | null>(
	null,
);

export function ProvideAuth({ children }: { children: React.ReactNode }) {
	const auth = useProvideAuth();
	return <authContext.Provider value={auth}>{children}</authContext.Provider>;
}
