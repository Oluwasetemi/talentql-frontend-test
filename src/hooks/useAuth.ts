import React from 'react';
import { authContext } from '../context/authContext';

export function useAuth() {
	return React.useContext(authContext);
}
