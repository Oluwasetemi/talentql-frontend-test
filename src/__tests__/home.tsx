import { render } from '@testing-library/react';
import 'jest-styled-components';
import * as React from 'react';
import { Provider } from 'react-redux';
import Home from '../components/Home';
import { authContext } from '../context/authContext';
import { useProvideAuth } from '../hooks/useProviderAuth';
import store from '../store';

const mockUseLocationValue = {
	pathname: '/',
	search: '',
	hash: '',
	state: null,
};

jest.mock('react-router-dom', () => ({
	...(jest.requireActual('react-router-dom') as {}),
	useLocation: jest.fn().mockImplementation(() => {
		return mockUseLocationValue;
	}),
}));

function HomeApp() {
	const auth = useProvideAuth();
	auth.signin(() => console.log('login'));
	return (
		<authContext.Provider value={auth}>
			<Provider store={store}>
				<Home />
			</Provider>
		</authContext.Provider>
	);
}

describe('<Home />', () => {
	test('home component should render a not logged in on render without auth', () => {
		const { getByText } = render(<HomeApp />);
		const input = getByText(/You are not logged in./i);
		expect(input).toMatchSnapshot();
	});
});
