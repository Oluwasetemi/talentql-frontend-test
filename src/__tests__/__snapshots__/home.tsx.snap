// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> home component should render a not logged in on render without auth 1`] = `
<p>
  You are not logged in.
</p>
`;
